### ISSUE TITLE
LoggingSystem should check the presence of log4j v2 before log4j v1

### ISSUE THREAD

**Issue by lite2073** [spring-projects/spring-boot#2274]
I'd like to use log4j v2 for my application which has to use some libraries that depend on log4j v1. So I include `org.apache.logging.log4j:log4j-1.2-api` dependency to support v1 in v2. However, LoggingSystem checks the presence of log4j v1 before log4j v2:
`LoggingSystem.java`
```
		systems.put("org.apache.log4j.PropertyConfigurator",
				"org.springframework.boot.logging.log4j.Log4JLoggingSystem");
		systems.put("org.apache.logging.log4j.LogManager",
				"org.springframework.boot.logging.log4j2.Log4J2LoggingSystem");
```

If we switch the order of the two statements, then it'd be able to detect log4j v2.

What do you think? Are there any workarounds?

**Comment by philwebb** [comment]
I've changed the order for 1.2.1 and also added system property support to allow a specific override. You might want to use the [Log4j 1.2 Bridge](http://logging.apache.org/log4j/2.x/log4j-1.2-api/) in your application to redirect any Log4J 1.2 calls to Log4J2.

**Comment by alwaysastudent** [comment]
The dependency definition for the same should be provided in the spring-boot-dependencies/pom.xml
Otherwise the consumer has to manage the version at their end which add to the redundancy and conflicts on upgrades.

```
<dependency>
    <groupId>org.apache.logging.log4j</groupId>
    <artifactId>log4j-1.2-api</artifactId>
    <version>${log4j2.version}</version>
</dependency>
```

**Comment by philwebb** [comment]
@alwaysastudent We generally only add dependency management for libraries that we use directly in Boot. Using the bridge shouldn't be that common so we recommend the `<version>${log4j2.version}</version>` as you've shown.

**Comment by alwaysastudent** [comment]
Right, but the consumers can not inherit the managed maven properties from your layer, unless they make yours as their parent pom. Most of us choose `import` pom. So in this case we have to duplicate the log4j2.version in our layer as well. 

From spring boot 1.3.x to 1.5.x, you have bumped up the log4j2.version from 2.4 to 2.7.x. The same has to happen in our layer as well with a conscious diligence. It would be nice to have that as well defined in your layer, since it is very common dependency.

**Comment by philwebb** [comment]
@alwaysastudent Please raise a new issue for this and we'll consider it some more.

**Comment by wilkinsona** [comment]
We're [using Log4J 2's bom in 1.5](https://github.com/spring-projects/spring-boot/commit/c30b917ba1d0e1d94c30d4964aa495c57aec6fe5) so we already have dependency management for the module in question.

**Comment by philwebb** [comment]
Great! Thanks @wilkinsona, no need for another issue @alwaysastudent.
